iaso/api/mobile/entity.py

```diff
@@ -197,3 +197,52 @@ def get_queryset(self):
             "attributes__form__form_versions",
         )
         return queryset.order_by("id")
+
+
+class DeletedMobileEntitySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Entity
+        fields = ["id", "uuid", "deleted_at", "merged_to_uuid"]
+
+    deleted_at = TimestampField()
+    merged_to_uuid = serializers.SerializerMethodField()
+
+    def get_merged_to_uuid(self, entity):
+        if entity.merged_to:
+            return entity.merged_to.uuid
+
+
+class MobileEntityDeletedViewSet(ModelViewSet):
+    f"""Entity API for mobile
+
+    list: /api/mobile/entities/deleted
+
+    Returns the full list of (soft-) deleted entities.
+    No pagination at the moment to keep thing simple.
+    """
+
+    results_key = "results"
+    remove_results_key_if_paginated = True
+    filter_backends = [filters.OrderingFilter, DjangoFilterBackend]
+    permission_classes = [
+        permissions.IsAuthenticated,
+        HasPermission(permission.ENTITIES),
+    ]  # type: ignore
+
+    def pagination_class(self):
+        return MobileEntitiesSetPagination(self.results_key)
+
+    def get_serializer_class(self):
+        return DeletedMobileEntitySerializer
+
+    def get_queryset(self):
+        user = self.request.user
+        app_id = self.request.query_params.get("app_id")
+
+        if not app_id:
+            raise ParseError("app_id is required")
```

Comment:
Nitpicking: There is a `AppIdSerializer` that retrieves the `app_id` from the query